python/cudf/cudf/core/column/column.py

```diff
@@ -242,13 +242,13 @@ def find_and_replace(
         raise NotImplementedError
 
     @acquire_spill_lock()
-    def clip(self, lo: ScalarLike, hi: ScalarLike) -> ColumnBase:
+    def clip(self, lo: ScalarLike, hi: ScalarLike) -> Self:
         plc_column = plc.replace.clamp(
             self.to_pylibcudf(mode="read"),
             cudf.Scalar(lo, self.dtype).device_value.c_value,
             cudf.Scalar(hi, self.dtype).device_value.c_value,
         )
-        return type(self).from_pylibcudf(plc_column)
+        return type(self).from_pylibcudf(plc_column)  # type: ignore[return-value]
```

Comment:
I suppose this is because `Column.from_pylibcudf` only advertises that it produces `ColumnBase` rather than `Self`? All this typing is kind of weird because the C++ column is type-erased.

Reply:
Yup exactly. We have this occurrences in quite a few places in cudf Python since we know what column subclass should be returned.